cpp/src_prims/metrics/precision_score.cuh

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <math.h>
+#include <raft/cudart_utils.h>
+#include <common/device_buffer.hpp>
+#include <cub/cub.cuh>
+#include <cuml/common/cuml_allocator.hpp>
+#include <raft/cuda_utils.cuh>
+
+namespace MLCommon {
+
+namespace Metrics {
+
+/**
+ * @brief kernel to calculate the precision score (currently only supporting binary class precision score)
+ * @param gold: 1D array-like of ground truth class labels
+ * @param pred: 1D array-like of predicted class labels
+ * @param size: the size of array a and b
+ * @param d_TP: pointer to the device memory that stores the aggregate true positive count
+ * @param d_FP: pointer to the device memory that stores the aggregate false positive count
+ */
+template <int BLOCK_DIM_X>
+__global__ void precision_score_kernel(const int *gold, const int *pred,
```

Comment:
This kernel can be simplified by quite a bit, and likely be much more performant. Are you aware of `__syncthreads_count` CUDA primitive usage?

Reply:
I will change it so that it uses `__syncthreads_count` - thanks!